packages/x-date-pickers/src/DesktopDatePicker/DesktopDatePicker.tsx

```diff
@@ -72,7 +72,7 @@ const DesktopDatePicker = React.forwardRef(function DesktopDatePicker<
     slotProps: {
       ...defaultizedProps.slotProps,
       field: (ownerState: any) => ({
-        ...resolveComponentProps(defaultizedProps.slotProps?.field, ownerState),
+        ...(resolveComponentProps(defaultizedProps.slotProps?.field, ownerState) as any),
```

Comment:
I'll try to get rid of this, it's caused by the introduction of `BaseSingleInputFieldPropsInPickerSlots` but I don't understand why